### PR TITLE
fix:Long console statement is truncated on the debugger

### LINKED ISSUE #16726: [Bug]: Long console statement is truncated on the debugger
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When console statement is long then the text is truncated when printed on the debugger as seen in the screenshot

### Steps To Reproduce

1. Add following statement to the editor and execute the function to observe truncation
`console.log('This is a long statement to display truncation issue present on the debugger. You can see this statement not being printed in whole')`

image

### Public Sample App

_No response_

### Version

Cloud

### PR DESCRIPTION
**Pull request by Shivam-z**
**Issue : Fix Truncated Logs for USER_GENERATED Events**

**Description:**  
This PR resolves an issue where **USER_GENERATED** logs were truncated in the debugger. We now conditionally display logs based on their category:

- For **USER_GENERATED** logs, the `logData` array is used to display the full message.
- For **PLATFORM_GENERATED** logs, the `text` prop is used as it's short and not truncated.

**Impact:**  
This fix ensures that longer log messages from user-generated events are fully displayed in the debugger, improving the logging experience without affecting system-generated logs.

---

**Before Fix**
Screenshot from 2024-09-19 09-47-57

**After Fix**
Screenshot from 2024-09-19 09-49-07

> [!TIP]  
> _Add a TL;DR when the description is longer than 500 words or extremely technical (helps the content, marketing, and DevRel team)._
>
> _Please also include relevant motivation and context. List any dependencies that are required for this change. Add links to Notion, Figma or any other documents that might be relevant to the PR._

Fixes #16726 
_or_  
Fixes `Issue URL`
> [!WARNING]  
> _If no issue exists, please create an issue first, and check with the maintainers if the issue is valid._

## Automation

/ok-to-test tags=""

### :mag: Cypress test results

> [!CAUTION]  
> If you modify the content in this section, you are likely to disrupt the CI result for your PR.

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [ ] No

## Summary by CodeRabbit

- **New Features**
	- Improved text display in the log items for better readability with line breaks and preserved whitespace.
	- Enhanced conditional rendering of log content based on log category, providing more relevant information to users.

### CODE REVIEW

**Comment by Shivam-z** [comment]
@appsmith-bot @ramsaptami please look into this PR and assign any reviewer to this else it will  be auto closed in some days.

File: app/client/src/components/editorComponents/Debugger/LogItem.tsx [hunk 1]
```diff
@@ -302,7 +302,9 @@ function LogItem(props: LogItemProps) {
               className="debugger-label t--debugger-log-message cursor-text"
               onClick={(e) => e.stopPropagation()}
             >
-              {props.text}
+              {props.category === LOG_CATEGORY.PLATFORM_GENERATED
+                ? props.text
+                : props.logData}
```
Comment (alex-golovanov) [inline app/client/src/components/editorComponents/Debugger/LogItem.tsx:307]: Hey @Shivam-z, first of all, thanks for contributing! I do have a concern, though. As coderabbitai pointed out, `props.logData` is an poorly typed array that could contain anything, including objects, in which case the app would crash. Even if it's just an array of strings we would need to make sure that message are printed in a readable fashion.

**Review by alex-golovanov — CHANGES_REQUESTED** [review]
Hi @Shivam-z, do you think you can address the issue with `logData` that I've outlined above? If not, pleas let me know, cheers.

File: app/client/src/components/editorComponents/Debugger/LogItem.tsx (hunk 1 above)
Comment (Shivam-z) [reply app/client/src/components/editorComponents/Debugger/LogItem.tsx:307]: Hi @alex-golovanov , thank you for your feedback, and I apologize for the delayed response. You're absolutely right `props.logData` is a poorly typed array that could indeed lead to issues, especially if it contains objects, potentially causing the app to crash.

In addition to that, I’ve investigated further and found another solution approach for this issue: the `log.text` prop, which is responsible for displaying the logs, is being [truncated](https://github.com/appsmithorg/appsmith/blob/a8da0e1e7232403665defdcd528aa93070b04b99/app/client/src/components/editorComponents/Debugger/helpers.tsx#L60) when the log length exceeds 100 characters. To resolve this, we could either ignore the truncation or increase the truncLen to ensure longer logs are displayed in full.

Let me know your thoughts on this approach.

Comment (alex-golovanov) [reply app/client/src/components/editorComponents/Debugger/LogItem.tsx:307]: @Shivam-z apologies for the late reply, let me check with the team to get more clarity on this.

**Comment by hetunandu** [comment]
Closing this because it is stale

**Pull request closed without merging**